fourier/fourier_example_test.go

```diff
@@ -0,0 +1,223 @@
+// Copyright ©2018 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package fourier_test
+
+import (
+	"fmt"
+	"math"
+	"math/cmplx"
+
+	"gonum.org/v1/gonum/floats"
+	"gonum.org/v1/gonum/fourier"
+	"gonum.org/v1/gonum/mat"
+)
+
+func ExampleFFT_FFT() {
+	// Period is a set of samples over a given period.
+	period := []float64{1, 0, 2, 0, 4, 0, 2, 0}
+
+	// Initialize an FFT and perform the analysis.
+	fft := fourier.NewFFT(len(period))
+	coeff := fft.FFT(nil, period)
+
+	for i, c := range coeff {
+		fmt.Printf("freq=%v cycles/period, magnitude=%v, phase=%.4g\n",
+			fft.Freq(i), cmplx.Abs(c), cmplx.Phase(c))
+	}
+
+	// Output:
+	//
+	// freq=0 cycles/period, magnitude=9, phase=0
+	// freq=0.125 cycles/period, magnitude=3, phase=3.142
+	// freq=0.25 cycles/period, magnitude=1, phase=-0
+	// freq=0.375 cycles/period, magnitude=3, phase=3.142
+	// freq=0.5 cycles/period, magnitude=9, phase=0
+}
+
+func ExampleFFT_FFT_tone() {
+	// Tone is a set of samples over a second of a pure Middle C.
+	const (
+		mC      = 261.625565 // Hz
+		samples = 44100
+	)
+	tone := make([]float64, samples)
+	for i := range tone {
+		tone[i] = math.Sin(mC * 2 * math.Pi * float64(i) / samples)
+	}
+
+	// Initialize an FFT and perform the analysis.
+	fft := fourier.NewFFT(samples)
+	coeff := fft.FFT(nil, tone)
+
+	var maxFreq, magnitude, mean float64
+	for i, c := range coeff {
+		m := cmplx.Abs(c)
+		mean += m
+		if m > magnitude {
+			magnitude = m
+			maxFreq = fft.Freq(i) * samples
+		}
+	}
+	fmt.Printf("freq=%v Hz, magnitude=%v, mean=%v\n", maxFreq, magnitude, mean/samples)
+
+	// Output:
+	//
+	// freq=262 Hz, magnitude=17296.195519181776, mean=2.783457755654771
+}
+
+func ExampleCmplxFFT_FFT() {
+	// Period is a set of samples over a given period.
+	period := []complex128{1, 0, 2, 0, 4, 0, 2, 0}
+
+	// Initialize a complex FFT and perform the analysis.
+	fft := fourier.NewCmplxFFT(len(period))
+	coeff := fft.FFT(nil, period)
+
+	for i := range coeff {
+		// Center the spectrum.
+		i = fft.ShiftIdx(i)
+
+		fmt.Printf("freq=%v cycles/period, magnitude=%v, phase=%.4g\n",
+			fft.Freq(i), cmplx.Abs(coeff[i]), cmplx.Phase(coeff[i]))
+	}
+
+	// Output:
+	//
+	// freq=-0.5 cycles/period, magnitude=9, phase=0
+	// freq=-0.375 cycles/period, magnitude=3, phase=3.142
+	// freq=-0.25 cycles/period, magnitude=1, phase=0
+	// freq=-0.125 cycles/period, magnitude=3, phase=3.142
+	// freq=0 cycles/period, magnitude=9, phase=0
+	// freq=0.125 cycles/period, magnitude=3, phase=3.142
+	// freq=0.25 cycles/period, magnitude=1, phase=0
+	// freq=0.375 cycles/period, magnitude=3, phase=3.142
+}
+
+func Example_fFT2() {
```

Comment:
Lower-case f in fFT2

Reply:
That's to give it package scope. See the example names section https://blog.golang.org/examples

Reply:
Oh, there's always something to learn. Thanks.